Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -871,3 +868,33 @@ theorem isTorsion_iff_isTorsion_int [AddCommGroup M] :
 #align add_monoid.is_torsion_iff_is_torsion_int AddMonoid.isTorsion_iff_isTorsion_int
 
 end AddMonoid
+
+namespace AddSubgroup
+
+variable (A : Type*) [AddCommGroup A] (n : ℕ)
+
+/-- The additive `n`-torsion subgroup for `n` in `ℕ`. -/
+@[reducible]
+def torsionBy := (Submodule.torsionBy ℤ A n).toAddSubgroup
+
+@[inherit_doc]
+scoped notation:max A"["n"]" => torsionBy A n
```

Comment:
This notation doesn't work with numerals, it seems:
```
ambiguous, possible interpretations 
  A[37] : AddSubgroup A
  
  A[37] : ?m.17
```
In general you might have better luck with some more unicodey square brackets. Perhaps better to get some other opinions on the Zulip? I'm reluctant to merge because I can't get the notation to work, but once this is fixed this is ready to go.

Reply:
The suggestion there is
```
scoped notation:max (priority := high) A"["n"]" => torsionBy A n
```
Does this solve the problems above?